presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/Decoder.java

```diff
@@ -13,12 +13,25 @@
  */
 package io.prestosql.elasticsearch.decoders;
 
+import io.prestosql.spi.PrestoException;
 import io.prestosql.spi.block.BlockBuilder;
 import org.elasticsearch.search.SearchHit;
 
 import java.util.function.Supplier;
 
+import static io.prestosql.spi.StandardErrorCode.TYPE_MISMATCH;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
 public interface Decoder
 {
     void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output);
+
+    default PrestoException cannotDecodeException(String type, Object value)
+    {
+        requireNonNull(type, "type is null");
+        requireNonNull(value, "value is null"); // null value must be handled explicitly by the caller
+
+        return new PrestoException(TYPE_MISMATCH, format("Unsupported representation for %s type: %s [%s]", value.getClass().getName(), value));
```

Comment:
We should also report the current column name.
otherwise its trial-n-err for users

Reply:
`value.getClass().getName()` is an artifact of how `SearchHit.getSourceAsMap()` parses and returns values, but it's not user-friendly or necessarily reflect anything directly under the control of the user. That's why the current code is designed to report the general expectation for the type, not the concrete Java type (e.g., "expected a list of elements", "expected a numeric value", or "expected an object", or "expected a string"-- these are all ES types).